src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,591 @@
+        self.print_step(22, "Send the Read Request Message to the DUT 3 timesand check if they are correct each time")
+        read_request_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [server_list_attr])
+        read_request_2 = await self.default_controller.ReadAttribute(self.dut_node_id, [server_list_attr])
+        read_request_3 = await self.default_controller.ReadAttribute(self.dut_node_id, [server_list_attr])
+        asserts.assert_equal(read_request_1, read_request_2)
+        asserts.assert_equal(read_request_2, read_request_3)
+
+        # Step 23
+        # TH sends a Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set.
+        # DUT sends back the attribute value with the DataVersion of the cluster.
+        # TH sends a second read request to the same cluster with the DataVersionFilter Field set with the dataversion value received before.
+        self.print_step(
+            23, "Send the Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set")
+        # Temporarily commented to avoid linter errors -- will revert once output value is known so that assertion can be used correctly
+        # read_request = await self.default_controller.ReadAttribute(self.dut_node_id, server_list_attr_path)
+        # data_version = read_request[0][Clusters.Descriptor][Clusters.Attribute.DataVersion]
+        # data_version_filter = [(0, Clusters.Descriptor, data_version)]
+
+        # read_request_2 = await self.default_controller.ReadAttribute(self.dut_node_id, server_list_attr_path, dataVersionFilters=data_version_filter)
+        # Seems to return {}?
+
+        # DUT should not send a report data action with the attribute value to the TH if the data version is same as that requested.
+
+        # Step 24
+        # TH sends a Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set.
+        # DUT sends back the attribute value with the DataVersion of the cluster.
+        # TH sends a write request to the same cluster to write to any attribute.
+        # TH sends a second read request to read an attribute from the same cluster with the DataVersionFilter Field set with the dataversion value received before.
+
+        self.print_step(
+            24, "Send the Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set")
+        # Temporarily commented to avoid linter errors -- will revert once output value is known so that assertion can be used correctly
+        # read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.Objects.Descriptor.Attributes.FeatureMap)])
+        await self.default_controller.WriteAttribute(self.dut_node_id, [(0, Clusters.Objects.Descriptor.Attributes.FeatureMap(value=123456))])
```

Comment:
This won't work - feature map is not writeable so you're not going to get a data version filter change. Try node label.

Reply:
Fixed in newest commits